ping.go

```diff
@@ -587,6 +590,17 @@ func (p *Pinger) sendICMP(conn *icmp.PacketConn) error {
 		}
 		p.PacketsSent++
 		p.sequence++
+
+		handler := p.OnSend
+		if handler != nil {
+			outPkt := &Packet{
+				Nbytes: len(msgBytes),
+				IPAddr: p.ipaddr,
+				Addr:   p.addr,
+				Seq:    p.sequence - 1,
+			}
+			handler(outPkt)
+		}
```

Comment:
If you swap around the order of execution it avoid doing an unnecessary `p.sequence - 1`:

```suggestion
		handler := p.OnSend
		if handler != nil {
			outPkt := &Packet{
				Nbytes: len(msgBytes),
				IPAddr: p.ipaddr,
				Addr:   p.addr,
				Seq:    p.sequence,
			}
			handler(outPkt)
		}

		p.PacketsSent++
		p.sequence++
```